src/server/common/controller/question-page-controller/question-page-controller.test.js

```diff
@@ -593,28 +582,21 @@ describe('QuestionPageController', () => {
         payload: { nextPage: 'test_next_page' }
       }
 
-      const result = controller.postHandler(postRequest, h)
+      controller.postHandler(postRequest, h)
 
-      expect(h.view).toHaveBeenCalledTimes(1)
-      expect(h.view).toHaveBeenCalledWith(
-        questionView,
-        expect.objectContaining({
+      expect(h.redirect).toHaveBeenCalledTimes(1)
+      expect(request.yar.clear).toHaveBeenCalledTimes(1)
+      expect(request.yar.set.mock.calls[1]).toEqual([
+        `errors:${sectionKey}:${questionKey}`,
+        {
           errorMessages: [
             { href: `#${questionKey}`, text: 'There is a problem' }
           ],
-          errors: { [questionKey]: { text: 'There is a problem' } },
-          viewModelOptions: { validate: true, question },
-          heading: question,
-          nextPage: 'test_next_page',
-          pageTitle: `Error: ${question}`,
-          value: undefined
-        })
-      )
-
-      const viewArgs = h.view.mock.calls[0][1]
-      expect(viewArgs.answer).toBeInstanceOf(TestAnswer)
-
-      expect(result).toBe('view')
+          errors: {
+            [questionKey]: { text: 'There is a problem' }
+          }
+        }
+      ])
```

Comment:
Aren't we also expecting this to persist the submitted payload (for replay during calls to the getHandler)?

Reply:
Also, 'toStrictEqual' is a little stronger of a matcher - see https://jestjs.io/docs/using-matchers